File: solutions/javascript/space-age/1/space-age.js
//
// This is only a SKELETON file for the 'Space Age' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

import { parse } from "path";

const ORBITAL_PERIODS = {
  mercury: 0.2408467,
  venus: 0.61519726,
  earth: 1,
  mars: 1.8808158,
  jupiter: 11.862615,
  saturn: 29.447498,
  uranus: 84.016846,
  neptune: 164.79132
}
export const age = (planet, seconds) => {
  return parseFloat((seconds / 31557600 / ORBITAL_PERIODS[planet]).toFixed(2));  
};

File: solutions/dart/space-age/1/lib/space_age.dart
class SpaceAge {
  num age({required String planet, required num seconds}) {
    num res;
    switch (planet) {
      case 'Earth':
        res = seconds / 31557600;
        break;
      case 'Mercury':
        res = seconds / 60191000;
        break;
      case 'Venus':
        res = seconds / 91959999;
        break;
      case 'Mars':
        res = seconds / 146000000;
        break;
      case 'Jupiter':
        res = seconds / 433281000;
        break;
      case 'Saturn':
        res = seconds / 1076023600;
        break;
      case 'Uranus':
        res = seconds / 3068800000;
        break;
      case 'Neptune':
        res = seconds / 6018300000;
        break;
      default:
        return 0;
    }
    return double.parse(res.toStringAsFixed(2));
  }
}
